### ISSUE TITLE
Can not update Private Key Password

### ISSUE THREAD

**Issue by gennari-dkt** [nextcloud/server#18742]
Hello,
I am using Nextcloud 17.0.2 after a migration from Owncloud 10.0. Nexcloud is installed in a Debian Buster machine with Nignx, postgres 10 and php 7.3. I set LDAP/AD integration.

A the moment I can not update private key password of Active Directory accounts. In the log files I found this error:

`{"reqId":"YCFKP5hzAB0RnbU3pORz","level":0,"time":"2020-01-08T10:43:56+00:00","remoteAddr":"10.39.0.58","user":"AACEE76A-E5A0-4983-B371-D82B4A255D8F","app":"user_ldap","method":"POST","url":"\/apps\/encryption\/ajax\/updatePrivateKeyPassword","message":{"Exception":"OCA\\User_LDAP\\Exceptions\\NotOnLDAP","Message":"No user available for the given login name on 10.39.0.202:389","Code":0,"Trace":[{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_LDAP.php","line":191,"function":"getLDAPUserByLoginName","class":"OCA\\User_LDAP\\User_LDAP","type":"->","args":["*** sensitive parameter replaced ***"]},{"function":"checkPassword","class":"OCA\\User_LDAP\\User_LDAP","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_Proxy.php","line":81,"function":"call_user_func_array","args":[[{"__class__":"OCA\\User_LDAP\\User_LDAP"},"checkPassword"],["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/Proxy.php","line":152,"function":"walkBackends","class":"OCA\\User_LDAP\\User_Proxy","type":"->","args":["*** sensitive parameter replaced ***","checkPassword",["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_Proxy.php","line":196,"function":"handleRequest","class":"OCA\\User_LDAP\\Proxy","type":"->","args":["*** sensitive parameter replaced ***","checkPassword",["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/User\/Manager.php","line":212,"function":"checkPassword","class":"OCA\\User_LDAP\\User_Proxy","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/User\/Manager.php","line":189,"function":"checkPasswordNoLogging","class":"OC\\User\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/encryption\/lib\/Controller\/SettingsController.php","line":114,"function":"checkPassword","class":"OC\\User\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":170,"function":"updatePrivateKeyPassword","class":"OCA\\Encryption\\Controller\\SettingsController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Encryption\\Controller\\SettingsController"},"updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/App.php","line":126,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Encryption\\Controller\\SettingsController"},"updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["SettingsController","updatePrivateKeyPassword",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/base.php","line":997,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/encryption\/ajax\/updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_LDAP.php","Line":176,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/79.0.3945.88 Safari\/537.36","version":"17.0.2.1"}`

Particularly the error message is "No user available for the given login name" and the user passed is AACEE76A-E5A0-4983-B371-D82B4A255D8F, the objectGUID of AD user, and not the username (sAMAccount name in AD).

Is it a bug or is it the normal behavior? 

Very kind regards.

Alessio

**Comment by kesselb** [comment]
Hey, please use the issue template to report a bug. Thanks 👍

**Comment by gennari-dkt** [comment]
### Steps to reproduce
Login
Go to Settings, security
Try to change Private key password


### Expected behaviour
Password is changed

### Actual behaviour
After pressing save password, the stare remain in "Saving....


## Server configuration detail

**Operating system:** Linux 4.19.0-6-amd64 #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64

**Webserver:** nginx/1.16.1 (fpm-fcgi)

**Database:** pgsql PostgreSQL 10.11 (Debian 10.11-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

**PHP version:** 

7.3.13-1+0~20191218.50+debian10~1.gbp23c2da
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, cgi-fcgi, PDO, xml, apcu, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, intl, json, ldap, exif, msgpack, pdo_pgsql, pgsql, apc, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Phar, memcached, Zend OPcache

**Nextcloud version:** 17.0.2 - 17.0.2.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Migration from owncloud 10

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.3.0
 - activity: 2.10.1
 - breezedark: 17.0.2
 - checksum: 0.4.3
 - cloud_federation_api: 1.0.0
 - comments: 1.7.0
 - dav: 1.13.0
 - encryption: 2.5.0
 - federatedfilesharing: 1.7.0
 - federation: 1.7.0
 - files: 1.12.0
 - files_external: 1.8.0
 - files_pdfviewer: 1.6.0
 - files_rightclick: 0.15.1
 - files_sharing: 1.9.0
 - files_snapshots: 0.3.0
 - files_trashbin: 1.7.0
 - files_versions: 1.10.0
 - files_videoplayer: 1.6.0
 - firstrunwizard: 2.6.0
 - gallery: 18.4.0
 - impersonate: 1.4.0
 - issuetemplate: 0.6.0
 - logreader: 2.2.0
 - lookup_server_connector: 1.5.0
 - metadata: 0.10.0
 - nextcloud_announcements: 1.6.0
 - notes: 3.1.1
 - notifications: 2.5.0
 - oauth2: 1.5.0
 - password_policy: 1.7.0
 - privacy: 1.1.0
 - provisioning_api: 1.7.0
 - recommendations: 0.5.0
 - serverinfo: 1.7.0
 - sharebymail: 1.7.0
 - support: 1.0.1
 - survey_client: 1.5.0
 - systemtags: 1.7.0
 - text: 1.1.1
 - theming: 1.8.0
 - twofactor_backupcodes: 1.6.0
 - twofactor_totp: 4.1.2
 - unsplash: 1.1.4
 - updatenotification: 1.7.0
 - user_ldap: 1.7.0
 - viewer: 1.2.0
 - workflowengine: 1.7.0
Disabled:
 - admin_audit

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "maintenance": false,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "10.39.5.5",
        "cloud.diakont.srl",
        "cloud.diakont.it"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/cloud.diakont.it",
    "dbtype": "pgsql",
    "version": "17.0.2.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "installed": true,
    "mail_smtpmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "25",
    "ldapIgnoreNamingRules": false,
    "mail_smtpauthtype": "PLAIN",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "loglevel": 0,
    "log_rotate_size": "104857600",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.distributed": "\\OC\\Memcache\\Memcached",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": true,
    "filelocking.ttl": 3600,
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
    "mail_sendmailmode": "smtp"
}
```
</details>

**Are you using external storage, if yes which one:** smb

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

<details><summary>LDAP configuration (delete this par if not used)</summary>

```
background_sync_interval: 43200background_sync_offset: 0background_sync_prefix: s01cleanUpJobOffset: 0enabled: yesinstalled_version: 1.7.0s01_lastChange: 1578579867s01has_memberof_filter_support: 1s01home_folder_naming_rule: s01last_jpegPhoto_lookup: 0s01ldap_agent_password: NDgpTil3dURrSDV4OFUsTA==s01ldap_attributes_for_group_search: s01ldap_attributes_for_user_search: s01ldap_backup_host: s01ldap_backup_port: s01ldap_base: DC=Diakont,DC=srls01ldap_base_groups: DC=Diakont,DC=srls01ldap_base_users: DC=Diakont,DC=srls01ldap_cache_ttl: 600s01ldap_configuration_active: 1s01ldap_display_name: displaynames01ldap_dn: CN=adview,OU=Service users,OU=Italia,OU=diakont.srl,DC=Diakont,DC=srls01ldap_dynamic_group_member_url: s01ldap_email_attr: mails01ldap_experienced_admin: 0s01ldap_expert_username_attr: s01ldap_expert_uuid_group_attr: s01ldap_expert_uuid_user_attr: objectguids01ldap_group_display_name: cns01ldap_group_filter: s01ldap_group_filter_mode: 0s01ldap_group_member_assoc_attribute: uniqueMembers01ldap_groupfilter_groups: s01ldap_groupfilter_objectclass: s01ldap_host: 10.39.0.202s01ldap_login_filter: (&(&(|(objectclass=person))(|(|(memberof=CN=owncloud,OU=User groups,OU=Italia,OU=diakont.srl,DC=Diakont,DC=srl)(memberof=CN=SP_owncloud,OU=russia,OU=diakont.srl,DC=Diakont,DC=srl))))(|(samaccountname=%uid)(|(mailPrimaryAddress=%uid)(mail=%uid))))s01ldap_login_filter_mode: 1s01ldap_loginfilter_attributes: s01ldap_loginfilter_email: 0s01ldap_loginfilter_username: 1s01ldap_nested_groups: 0s01ldap_override_main_server: s01ldap_paging_size: 500s01ldap_port: 389s01ldap_quota_attr: s01ldap_quota_def: s01ldap_tls: 0s01ldap_turn_off_cert_check: 0s01ldap_user_display_name_2: s01ldap_user_filter_mode: 1s01ldap_user_name: samaccountnames01ldap_userfilter_groups: s01ldap_userfilter_objectclass: persons01ldap_userlist_filter: (&(&(|(objectclass=person))(|(|(memberof=CN=owncloud,OU=User groups,OU=Italia,OU=diakont.srl,DC=Diakont,DC=srl)(memberof=CN=SP_owncloud,OU=russia,OU=diakont.srl,DC=Diakont,DC=srl))))(!(userAccountControl:1.2.840.113556.1.4.803:=2)))s01use_memberof_to_detect_membership: 1signed: truetypes: authentication
```
</details>

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.117 Safari/537.36

**Operating system:** Windows 7, Windows 10

## Logs

<details><summary>Web server error log</summary>

```
{"reqId":"YCFKP5hzAB0RnbU3pORz","level":0,"time":"2020-01-08T10:43:56+00:00","remoteAddr":"10.39.0.58","user":"AACEE76A-E5A0-4983-B371-D82B4A255D8F","app":"user_ldap","method":"POST","url":"\/apps\/encryption\/ajax\/updatePrivateKeyPassword","message":{"Exception":"OCA\\User_LDAP\\Exceptions\\NotOnLDAP","Message":"No user available for the given login name on 10.39.0.202:389","Code":0,"Trace":[{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_LDAP.php","line":191,"function":"getLDAPUserByLoginName","class":"OCA\\User_LDAP\\User_LDAP","type":"->","args":["*** sensitive parameter replaced ***"]},{"function":"checkPassword","class":"OCA\\User_LDAP\\User_LDAP","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_Proxy.php","line":81,"function":"call_user_func_array","args":[[{"__class__":"OCA\\User_LDAP\\User_LDAP"},"checkPassword"],["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/Proxy.php","line":152,"function":"walkBackends","class":"OCA\\User_LDAP\\User_Proxy","type":"->","args":["*** sensitive parameter replaced ***","checkPassword",["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_Proxy.php","line":196,"function":"handleRequest","class":"OCA\\User_LDAP\\Proxy","type":"->","args":["*** sensitive parameter replaced ***","checkPassword",["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/User\/Manager.php","line":212,"function":"checkPassword","class":"OCA\\User_LDAP\\User_Proxy","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/User\/Manager.php","line":189,"function":"checkPasswordNoLogging","class":"OC\\User\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/encryption\/lib\/Controller\/SettingsController.php","line":114,"function":"checkPassword","class":"OC\\User\\Manager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":170,"function":"updatePrivateKeyPassword","class":"OCA\\Encryption\\Controller\\SettingsController","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Encryption\\Controller\\SettingsController"},"updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/App.php","line":126,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Encryption\\Controller\\SettingsController"},"updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["SettingsController","updatePrivateKeyPassword",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"encryption.Settings.updatePrivateKeyPassword"}]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/lib\/base.php","line":997,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/encryption\/ajax\/updatePrivateKeyPassword"]},{"file":"\/usr\/share\/nginx\/html\/nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/usr\/share\/nginx\/html\/nextcloud\/apps\/user_ldap\/lib\/User_LDAP.php","Line":176,"CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/79.0.3945.88 Safari\/537.36","version":"17.0.2.1"}
```
</details>

<details><summary>Nextcloud log</summary>

```

```
</details>

<details><summary>Browser log</summary>


</details>

**Comment by gennari-dkt** [comment]
Here is to you.
